src/main/java/org/example/controller/StudentController.java

```diff
@@ -0,0 +1,64 @@
+package org.example.controller;
+
+import org.example.model.Student;
+import org.example.service.StudentService;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.Collection;
+import java.util.Collections;
+
+@RestController
+@RequestMapping("/student")
+public class StudentController {
+
+    private final StudentService studentService;
+
+    public StudentController(StudentService studentService) {
+        this.studentService = studentService;
+    }
+
+    @GetMapping("{id}")
+    public ResponseEntity<Student> getStudent(@PathVariable Long id) {
+        Student student = studentService.findStudent(id);
+        if (student == null) {
+            return ResponseEntity.notFound().build();
+        }
+        return ResponseEntity.ok(student);
+    }
+
+    @GetMapping("/all")
+    public ResponseEntity<Collection<Student>> getAllStudents() {
+        return ResponseEntity.ok(studentService.getAllStudents());
+    }
+
+    @GetMapping("/age")
+    public ResponseEntity<Collection<Student>> getStudentsByAge(@RequestParam(required = false) int age) {
+        if (age > 0) {
+            return ResponseEntity.ok(studentService.getFilteredStudentsByAge(age));
+        }
+        return ResponseEntity.ok(Collections.emptyList());
+    }
+
+    @PostMapping()
+    public Student createStudent(@RequestBody Student student) {
+        return studentService.createStudent(student);
+    }
+
+    @PutMapping
+    public ResponseEntity<Student> editStudent(@RequestBody Student student) {
+        Student foundeStudent = studentService.editStudent(student);
+        if (foundeStudent == null) {
+            return ResponseEntity.status(HttpStatus.BAD_REQUEST).build();
+        }
+        return ResponseEntity.ok(foundeStudent);
+    }
+
+    @DeleteMapping("{id}")
+    public ResponseEntity deleteStudent(@PathVariable Long id) {
```

Comment:
```suggestion
    public ResponseEntity<?> deleteStudent(@PathVariable Long id) {
```
дженерик надо писать